### PR TITLE
Unify book morale bonuses from spiritual/cannibal into book_fun_for

### PR DESCRIPTION
**Pull request by Inglonias**
#### Summary
```SUMMARY: Interface "Show fun bonuses from spiritual books in read menu"```

#### Purpose of change
Show the fun boost in the Reading menu for a character who is spiritual (or a cannibal)

#### Describe the solution
The fun bonuses for these traits were moved into `book_fun_for`, and numbers were adjusted to keep behavior the same for all existing books. Fun is now adjusted as follows:

Spiritual characters reading a book with the INSPIRATIONAL flag get `abs(base_fun * 3)` fun from the book.

Cannibals/psychopaths/sapiovores reading "To Serve Man" get `abs(base_fun * 5)` fun from the book.

Note that the fun bonus from LOVES_BOOKS and HATES_BOOKS is boosted before this occurs, meaning that **a Spiritual character that also Hates Books will get no fun from reading inspirational books, and the same is true for cannibals reading To Serve Man.**

#### Describe alternatives you've considered
Leaving the fun bonuses where they were is totally an option. I also considered putting the behavior for "To Serve Man" into a flag so we can extend it, but decided to leave that up to whoever wants to add a post-cataclysm religious tome about eating people.

#### Additional context
I also edited the code for NPCs in a reading session, so that if any of them have traits, they should also keep their bonuses from those traits.

Book_fun_for now requires a player object to be passed to it. I passed it by reference, but there's one or two instances where I pass it by using a `*this`, and I'm not sure how good or bad an idea that actually is.

This is my second code contribution related to books. I'm hoping it's also my last, as I'd love to get to know other aspects of the code besides the part that covers what happens when you read a friggin' book!

### CODE REVIEW

**Comment by Inglonias** [comment]
i don't know how astyle works on windows plz halp.

**Comment by Inglonias** [comment]
Merge conflicts were the result of overzealous AStyle usage, and that has been fixed.

File: src/player.cpp [hunk 1]
```diff
@@ -9166,26 +9166,38 @@ bool player::fun_to_read( const item &book ) const
...
+    // If you don't have a problem with eating humans, To Serve Man becomes rewarding
+    if( ( p.has_trait( trait_CANNIBAL ) || p.has_trait( trait_PSYCHOPATH ) ||
+          p.has_trait( trait_SAPIOVORE ) ) &&
+        book.typeId() == "cookbook_human" ) {
+        fun_bonus = abs( fun_bonus );
+    } else if( p.has_trait( trait_SPIRITUAL ) && book.has_flag( "INSPIRATIONAL" ) ) {
+        fun_bonus = abs( fun_bonus * 3 );
+    }
+
+
```
Comment (Night-Pryanik) [inline src/player.cpp]: Empty lines.

File: src/player.cpp [hunk 2]
```diff
@@ -9472,8 +9477,10 @@ void player::do_read( item &book )
         if (reading->intel != 0) {
             add_msg(m_info, _("Requires intelligence of %d to easily read."), reading->intel);
         }
-        if (book_fun_for( book ) != 0) {
-            add_msg(m_info, _("Reading this book affects your morale by %d"), book_fun_for( book ) );
+        //It feels wrong to use a pointer to *this, but I can't find any other player pointers in this method.
+        if( book_fun_for( book, *this ) != 0 )
+        {
```
Comment (Night-Pryanik) [inline src/player.cpp]: Raise curly bracket one line higher, so it should look like `if( book_fun_for( book, *this ) != 0 ) {`

File: src/player.cpp (hunk 1 above)
Comment (Inglonias) [reply src/player.cpp]: Done...

File: src/player.cpp (hunk 2 above)
Comment (Inglonias) [reply src/player.cpp]: ...and done.

File: src/player.cpp [hunk 3]
```diff
@@ -9166,26 +9166,36 @@ bool player::fun_to_read( const item &book ) const
...
         debugmsg( "called player::book_fun_for with non-book" );
         return 0;
     }
+
     if( has_trait( trait_LOVES_BOOKS ) ) {
-        return ( book.type->book->fun + 1 );
-    } else if ( has_trait( trait_HATES_BOOKS ) ) {
+        fun_bonus++;
+    } else if( has_trait( trait_HATES_BOOKS ) ) {
         if( ( book.type->book->fun > 0 ) ) {
```
Comment (Night-Pryanik) [inline src/player.cpp]:
```suggestion
        if( book.type->book->fun > 0 ) {
```

File: src/player.cpp [hunk 4]
```diff
@@ -9521,36 +9525,10 @@ void player::do_read( item &book )
...
-                book.typeId() == "cookbook_human" ) {
-                fun_bonus = 25;
-                learner->add_morale( MORALE_BOOK, fun_bonus, fun_bonus * 3, 6_minutes, 3_minutes, true, book.type );
-            } else if( learner->has_trait( trait_SPIRITUAL ) && book.has_flag( "INSPIRATIONAL" ) ) {
-                fun_bonus = 15;
-                learner->add_morale( MORALE_BOOK, fun_bonus, fun_bonus * 5, 9_minutes, 9_minutes, true, book.type );
-            } else {
-                learner->add_morale( MORALE_BOOK, fun_bonus, book_fun_for( book ) * 15, 6_minutes, 3_minutes, true, book.type );
-            }
+        if ( book_fun_for( book, *learner ) != 0 ) {
```
Comment (Night-Pryanik) [inline src/player.cpp]:
```suggestion
        if( book_fun_for( book, *learner ) != 0 ) {
```

Comment (Inglonias) [reply src/player.cpp]: Yeah, whatever.

**Pull request merged**